### PR TITLE
New single query post-processor

### PR DESCRIPTION
**Pull request by jhoenicke**
Changed the post-processor of single-query track to be more strict.  It no longer ignores output before the first sat/unsat output.  Success is still ignored for backwards-compatibility with old scramblers and comments/white-space is allowed, but everything else triggers unknown result.

Note everything after sat/unsat is ignored, even if it is on the same line.

I added a few more unit test and moved one that is now sat instead of unknown.

### CODE REVIEW

File: single-query-track/process [hunk 1]
```diff
@@ -14,10 +15,10 @@
 STATUS=$(
 # remove the StarExec timing information from each line of the solver output
     sed 's/^[0-9]*\.[0-9]*\/[0-9]*.[0-9]*\t//g' "$1" | \
-# filter out lines containing only sat, unsat, or unknown
-    egrep -e '^\s*(sat|unsat|unknown)\s*$' | \
+# remove empty lines and lines containing only success or a comment
+    egrep -v -e '^\s*(success|;.*)?\s*$' | \
 # the first answer wins
-    head -1 | egrep -o '(sat|unsat|unknown)')
+    head -1 | sed 's/^\s*\([a-z]\+\)\b.*/\1/')
```
Comment (aehyvari) [inline single-query-track/process:21]: This won't work in OS X.  I'm not sure what's the right way around it though.  Either we don't care, or we use gsed in os X?

**Comment by HanielB** [comment]
I forget... what was our motivation, if any, to keep ignoring the output after the first line? Maybe if we're gonna be strict with nothing other than comments/success before result, we might want to be that struct for after as well?

**Comment by jhoenicke** [comment]
I updated the script to be compatible with sed.

**Review by aehyvari — APPROVED** [review]
To me this looks good now!

**Pull request merged**